### ISSUE TITLE
FR: Tube lights for projectiles

### ISSUE THREAD

**Issue by Kestrellius** [scp-fs2open/fs2open.github.com#4238]
Following on from configurable lights for weapons by @EatThePath, it would be useful to have configurable tube lights for standard projectiles like lasers and missiles, rather than being restricted to point lights.

Useful parameters include:

Position of head and tail ends of tube light (relative to projectile location; it's useful to be able to extend the light both forward and back from the point where the projectile exists). Note: The obvious necessity is position along the axis of movement. However, while I don't have any use cases in mind right now, the ability to position the head and tail along the other two axes would be a good thing to have, so if it's easy to add please do so.

Color, intensity, and radius of the light at its tail end.

**Comment by EatThePath** [comment]
The two hardest parts of this are probably the idea of variable properties from head to tail, and avoiding having lights stick out the back of a fighter/turret in the first moments. when long lights are used. Both should be managable, just making sure I think about them when I get back to it.

I'll have to review what the graphics options do currently but ideally I'd set this up so max light detail enables this, and max-1 forces all projectiles to be points. I don't expect this to be ruinious on performance but it seems prudent to allow for that possibility

**Comment by z64555** [comment]
Oof... would it be possible to use point lights with the start and end colors.  IIRC at some point the tube lights used multiple point lights along a line, so interpolation of the color of the point light could be done in parallel or more ideally nab the color of the bolt at its associated position.

**Comment by The-E** [comment]
@Kestrellius Can you elaborate on the effect that you want to achieve with this? Off the top of my head, I can only think of very few scenarios where something like this would make actual sense (like, for example, a primary shot with a very long projectile), and none where it would make _practical_ sense. The thing is that tube lights _are_ more expensive than point lights, and using them for something big and attention-grabbing as a beam makes a lot of sense (especially given that beams tend to be much rarer than projectiles), but projectile weapons in FSO are generally more numerous and short-lived.

**Comment by Kestrellius** [comment]
I have lasers with very long, very fast projectiles. I want to give them very intense lights because it produces a desirable effect, but I'm trying to keep the radius fairly small so that it's not too overwhelming. The issue is that once the radius gets small enough, the bolt travels more than the diameter of the light in a single frame, which produces a very unnatural-looking sort of discontinuity.

So in other words, I need tube lights to simulate motion blur.

**Comment by EatThePath** [comment]
The request got my attention because some of the warmachine projectiles are similar, and also because if someone wanted to use a single sprite to depict multiple shots in a burst of fire from a very high rate of fire weapon it might make sense there.
